python/ray/_private/resource_isolation_config.py

```diff
@@ -0,0 +1,242 @@
+    Also, converts system_reserved_cpu into cpu.weights for cgroupv2.
+
+    Raises:
+        ValueError on invalid inputs.
+
+    Attributes:
+        enable_resource_isolation: True if cgroupv2 based isolation of ray
+            system processes is enabled.
+        cgroup_path: The path for the cgroup the raylet should use to enforce
+            resource isolation.
+        system_reserved_cpu: The amount of cores reserved for ray system
+            processes. Must be >= ray_constants.MINIMUM_SYSTEM_RESERVED_CPU_CORES
+            and < the total number of cores available.
+        system_reserved_memory: The amount of memory in bytes reserved
+            for ray system processes. Must be >= ray_constants.MINIMUM_SYSTEM_RESERVED_MEMORY_BYTES
+            and system_reserved_cpu + object_store_bytes < the total memory available.
+    """
+
+    def __init__(
+        self,
+        enable_resource_isolation: bool = False,
+        cgroup_path: Optional[str] = None,
+        system_reserved_cpu: Optional[float] = None,
+        system_reserved_memory: Optional[int] = None,
+    ):
+
+        self._resource_isolation_enabled = enable_resource_isolation
+        self.cgroup_path = cgroup_path
+        self.system_reserved_memory = system_reserved_memory
+        # cgroupv2 cpu.weight calculated from system_reserved_cpu
+        # assumes ray uses all available cores
+        self.system_reserved_cpu_weight: int = None
+        # used in add_object_store_memory to ensure
+        # object_store_memory is not added twice to the
+        # number of bytes reserved for system processes
+        self._constructed = False
+
+        if not enable_resource_isolation:
+            if system_reserved_cpu:
+                raise ValueError(
+                    "system_reserved_cpu cannot be set when resource isolation is not enabled. "
+                    "Set enable_resource_isolation to True if you're using ray.init or use the "
+                    "--enable-resource-isolation flag if you're using the ray cli."
+                )
+
+            if self.system_reserved_memory:
+                raise ValueError(
+                    "system_reserved_memory cannot be set when resource isolation is not enabled. "
+                    "Set enable_resource_isolation to True if you're using ray.init or use the "
+                    "--enable-resource-isolation flag if you're using the ray cli."
+                )
+            if self.cgroup_path:
+                raise ValueError(
+                    "cgroup_path cannot be set when resource isolation is not enabled. "
+                    "Set enable_resource_isolation to True if you're using ray.init or use the "
+                    "--enable-resource-isolation flag if you're using the ray cli."
+                )
+            return
+
+        # resource isolation is enabled
+        self.system_reserved_cpu_weight = self._validate_system_reserved_cpu(
```

Comment:
nit: it reads to me, the word validates you only do validation/checking, but not side effect or return value, what about "validate_and_get_xxx"?